src/game_state/GameObject.h

```diff
@@ -19,6 +19,8 @@ class GameObject {
   virtual ~GameObject();
 
   virtual std::shared_ptr<Shape> GetModel() = 0;
+  virtual bool isCollectible() = 0;
+  virtual bool isObstacle() = 0;
```

Comment:
These should be replaced with a single method that returns an enum or enum class